### PR TITLE
TDP: add mouse scroll support

### PR DESCRIPTION
**Pull request by awly**
Add new message type to RFD - mouse wheel.
Implement the TDP encoder/decoder and conversion to RDP messages.

### CODE REVIEW

File: lib/srv/desktop/rdp/rdpclient/Cargo.toml [hunk 1]
```diff
@@ -21,5 +21,5 @@ num-traits = "0.2"
 # RustCrypto doesn't expose the low-level primitives we need for the smartcard
 # challenge signing (see src/piv.rs for details).
 openssl = "0.10.36"
-rdp-rs = { git = "https://github.com/gravitational/rdp-rs", rev = "92a80a25e27a802ffb2d21ac9d628e27e831dd5c" }
+rdp-rs = { git = "https://github.com/gravitational/rdp-rs", branch = "andrew/mouse-scroll" }
```
Comment (awly) [inline lib/srv/desktop/rdp/rdpclient/Cargo.toml:24]: TODO(me): change this to a master `rev` after https://github.com/gravitational/rdp-rs/pull/3 gets merged

File: Makefile [hunk 2]
```diff
@@ -245,6 +245,8 @@ ifeq ("$(DESKTOP_ACCESS)", "yes")
 .PHONY: rdpclient
 rdpclient:
 	cargo build --manifest-path=lib/srv/desktop/rdp/rdpclient/Cargo.toml --release
+	cargo install cbindgen
```
Comment (zmb3) [inline Makefile:248]: This is unrelated to mouse scroll and just something we need to get in, right?

File: lib/srv/desktop/rdp/rdpclient/client.go [hunk 3]
```diff
@@ -229,6 +230,40 @@ func (c *Client) start() {
...
+				}
+			case deskproto.MouseWheel:
+				var wheel C.CGOPointerWheel
+				switch m.Axis {
+				case deskproto.VerticalWheelAxis:
+					wheel = C.PointerWheelVertical
+				case deskproto.HorizontalWheelAxis:
+					wheel = C.PointerWheelHorizontal
+					// Fix the scroll direction to match RFD, it's inverted for
+					// horizontal scroll in RDP.
```
Comment (zmb3) [inline lib/srv/desktop/rdp/rdpclient/client.go:247]: huh?

File: lib/srv/desktop/rdp/rdpclient/src/lib.rs [hunk 4]
```diff
@@ -361,6 +361,8 @@ pub struct CGOPointer {
     pub y: u16,
     pub button: CGOPointerButton,
     pub down: bool,
+    pub wheel: CGOPointerWheel,
+    pub wheel_delta: i16,
```
Comment (zmb3) [inline lib/srv/desktop/rdp/rdpclient/src/lib.rs:365]: Why `i16` when its an `int32` in Go?

File: lib/srv/desktop/deskproto/proto.go [hunk 5]
```diff
@@ -307,6 +310,43 @@ func decodeClientUsername(in peekReader) (ClientUsername, error) {
...
+const (
+	VerticalWheelAxis   = MouseWheelAxis(0)
+	HorizontalWheelAxis = MouseWheelAxis(1)
+)
+
+// MouseWheel is the mouse wheel scroll message.
+// https://github.com/gravitational/teleport/blob/master/rfd/0037-desktop-access-protocol.md#8---mouse-wheel
+type MouseWheel struct {
+	Axis  MouseWheelAxis
+	Delta int32
```
Comment (zmb3) [inline lib/srv/desktop/deskproto/proto.go:325]: Not a huge deal, but I would consider putting delta first to ensure aligned reads and potentially save on padding if we expand this message in the future.

**Review by zmb3 — APPROVED** [review]

**Review by ibeckermayer — APPROVED** [review]

**Review by russjones — APPROVED** [review]
Bot.
